scene/main/canvas_item.cpp

```diff
@@ -92,11 +96,13 @@ void CanvasItem::set_visible(bool p_visible) {
 	visible = p_visible;
 	RenderingServer::get_singleton()->canvas_item_set_visible(canvas_item, p_visible);
 
-	if (!is_inside_tree()) {
+	if (!parent_visible_in_tree) {
+		notification(NOTIFICATION_VISIBILITY_CHANGED);
 		return;
 	}
```

Comment:
Previously, it won't emit `visibility_changed` when the `CanvasItem` is not in tree, after this change it will. I don't think this is expected.

Reply:
I am aware of that. I don't think this is undesired, previously if node was not visible in tree it would emit the signal too, which makes as much sense (but is actually needed for scene tree dock).